### ISSUE TITLE
Allow passing certificates as strings

### ISSUE THREAD

**Issue by jbaron-gingco** [guzzle/guzzle#2036]
| Q            | A
| ------------ | ---
| Bug?         | no
| New Feature? | yes
| Version      | 6.3.2 on Ubuntu 17.10 with self-built PHP 7.1.5

TLDR: It would be nice if PEM data could be passed directly as string in the  `cert` and `ssl_key` options.

The "cert" and "ssl_key" options currently require a path to a PEM encoded file as value. This requires that the certificate and its key are stored in a locally accessible storage. Sometimes one does not want that, but only wants to hold certificates in memory (e.g. they are retrieved from some configuration distribution service, from an environment variable, or similar).

It does not seem possible to pass the certificates as a string. I've played around with stream wrappers (e.g. using a file name like `data://text/plain;base64,SSBsb3ZlIFBIUAo=`), but that results in a "file not found" error. This error seems to come directly from OpenSSL or curl.

**Comment by sagikazarmark** [comment]
The problem is that libcurl has no API for passing keys as strings (only as files). So this solution would require Guzzle to temporarily store them as files which would involve a lot of checks, cleanups, etc and it's not something that Guzzle should do.

I'm afraid this is not something that we can implement in Guzzle right now.

**Comment by LeoMachado94** [comment]
Isso já é possível hoje @sagikazarmark ? Estou querendo passar como string..
